programs/launcher/main.cpp

```diff
@@ -1210,48 +1214,91 @@ launcher_def::kill (launch_modes mode, string sig_opt) {
   }
 }
 
-void
-launcher_def::bounce (const string& node_numbers) {
-   vector<string> nodes;
-   boost::split(nodes, node_numbers, boost::is_any_of(","));
-   for (string node_number: nodes) {
-      uint16_t node = -1;
-      try {
-         node = boost::lexical_cast<uint16_t,string>(node_number);
+vector<pair<host_def, eosd_def>>
+launcher_def::get_nodes(const string& node_number_list) {
+   vector<pair<host_def, eosd_def>> node_list;
+   string nnl = node_number_list;
+   std::transform(nnl.begin(), nnl.end(), nnl.begin(), ::tolower);
+   if (nnl == "all") {
+      for (auto host: bindings) {
+         for (auto node: host.instances) {
```

Comment:
Performance doesn't matter here, but `const auto& x` is a better habit to get into.